### ISSUE TITLE
Use error return value instead of embedding the error in the message

### ISSUE THREAD

**Issue by dhiller** [kubevirt/kubevirt#3127]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What happened**:
In a PR discussion we discovered [Stringly typed errors](https://github.com/kubevirt/kubevirt/pull/3039/files#r388228965), which should be cleaned up.

**What you expected to happen**:
Instead of using the message to convey the error only, also return the error with the corresponding return value.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by petrkotas** [comment]
Thank you @dhiller for creating this issue. I will work on it as soon as possible.

**Comment by slintes** [comment]
I replied on the issue, let me repeat here:

The grpc/protobuf implementation generates the func signature with err even though we don't declare it (see cmd.proto and cmd.pb.go). Also, it uses the error in case something goes wrong during the grpc call itself.
We, on the other hand, only use our own declared response type for reporting non-grpc related errors. It is stringly typed for sure, but IIRC it has always been this way, even in the pre-grpc implementation. If not, blame me, I introduced grpc a while ago ;) However, I don't say they is no room for improvement, just wanted give some more context :)

If you work on this, keep in mind that virt-handler needs to be able to communicate with an older virt-launcher (Kubevirt is updated but VMs keep running)
